Panic: runtime error: index out of range，

i do not know what happend,my influxdb crash too many times today.every time it only work fews minutes.every time is the same error.
influxdb:1.8.0 

ts=2020-12-03T04:27:28.805139Z lvl=info msg="Retention policy deletion check (start)" log_id=0QqgnOFG000 service=retention trace_id=0QqiXAeG000 op_name=retention_delete_check op_event=start
ts=2020-12-03T04:27:28.811008Z lvl=info msg="Deleted shard group" log_id=0QqgnOFG000 service=retention trace_id=0QqiXAeG000 op_name=retention_delete_check db_instance=telegraf db_shard_group=895 db_rp=umcs_policy
ts=2020-12-03T04:27:28.812257Z lvl=info msg="Deleted shard group" log_id=0QqgnOFG000 service=retention trace_id=0QqiXAeG000 op_name=retention_delete_check db_instance=telegraf_thd db_shard_group=896 db_rp=umcs_policy
ts=2020-12-03T04:27:28.851191Z lvl=info msg="Deleted shard" log_id=0QqgnOFG000 service=retention trace_id=0QqiXAeG000 op_name=retention_delete_check db_instance=telegraf_thd db_shard_id=896 db_rp=umcs_policy
panic: runtime error: index out of range [1] with length 0

goroutine 829 [running]:
encoding/binary.bigEndian.Uint16(...)
	/usr/local/go/src/encoding/binary/binary.go:100
github.com/influxdata/influxdb/tsdb.ReadSeriesKeyMeasurement(...)
	/go/src/github.com/influxdata/influxdb/tsdb/series_file.go:380
github.com/influxdata/influxdb/tsdb.parseSeriesKey(0x7f89985fd001, 0x1, 0x7f1fff, 0xc00b6700f0, 0x5, 0x5, 0xc00b6700f0, 0x5, 0x5, 0xc010c0a000, ...)
	/go/src/github.com/influxdata/influxdb/tsdb/series_file.go:417 +0x534
github.com/influxdata/influxdb/tsdb.ParseSeriesKeyInto(...)
	/go/src/github.com/influxdata/influxdb/tsdb/series_file.go:408
github.com/influxdata/influxdb/tsdb.(*Store).DeleteShard.func3(0x3c700)
	/go/src/github.com/influxdata/influxdb/tsdb/store.go:771 +0xcf
github.com/influxdata/influxdb/tsdb.(*SeriesIDSet).ForEach(0xc00e7174a0, 0xc016a171e8)
	/go/src/github.com/influxdata/influxdb/tsdb/series_set.go:181 +0xb3
github.com/influxdata/influxdb/tsdb.(*Store).DeleteShard(0xc000423200, 0x37b, 0x0, 0x0)
	/go/src/github.com/influxdata/influxdb/tsdb/store.go:765 +0x5ca
github.com/influxdata/influxdb/services/retention.(*Service).run(0xc00001b5e0)
	/go/src/github.com/influxdata/influxdb/services/retention/service.go:134 +0x1ff
github.com/influxdata/influxdb/services/retention.(*Service).Open.func1(0xc00001b5e0)
	/go/src/github.com/influxdata/influxdb/services/retention/service.go:51 +0x61
created by github.com/influxdata/influxdb/services/retention.(*Service).Open
	/go/src/github.com/influxdata/influxdb/services/retention/service.go:51 +0x1a9

--- a/tsdb/series_segment.go
+++ b/tsdb/series_segment.go
@@ -212,7 +212,10 @@ func (s *SeriesSegment) Flush() error {
 	if s.w == nil {
 		return nil
 	}
-	return s.w.Flush()
+	if err := s.w.Flush(); err != nil {
+		return err
+	}
+	return s.file.Sync()
 }
 
 // AppendSeriesIDs appends all the segments ids to a slice. Returns the new slice.

